bin/bl-exit

```diff
@@ -1,154 +1,425 @@
 #!/usr/bin/env python2.7
 # -*- coding: utf-8 -*-
+from __future__ import print_function
 
-import gtk
-import pygtk
-pygtk.require('2.0')
+import os
+display = os.environ.get('DISPLAY') is not None
 
+import sys
 import getopt
 import getpass
 import subprocess
-import sys
+import dbus
+
+__me__ = 'bl-exit'
+__version__ = '2.0.0'
+
+# Default errorMessages dict
+# key = result from _can_do_action
+# value = string shown in error_message_dialog
+canDoItErrorMessages = dict(
+    no="Action is not allowed.",
+    na="Action is not available.",
+    challenge="Action is not authorized."
+)
+
+# Translate command-line option to method - command line only
+actionToMethod = dict(
+    logout='Logout', l='Logout',
+    suspend='Suspend', s='Suspend',
+    hybridsleep='HybridSleep', y='HybridSleep',
+    hibernate='Hibernate', i='Hibernate',
+    reboot='Reboot', b='Reboot',
+    poweroff='PowerOff', p='PowerOff'
+)
+
+
+class CanDoItError(Exception):
+    pass
+
+
+class BlexitBase(object):
```

Comment:
These long class definitions are hard to read. Maybe it's time to put them into a separate module and just import them.